### ISSUE TITLE
deprecated function warning with elixir 1.10.1/OTP 22

### ISSUE THREAD

**Issue by bryanhuntesl** [bugsnag-elixir/bugsnag-elixir#94]
```
% iex -S mix   
Erlang/OTP 22 [erts-10.6] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [hipe]

warning: :restart and :shutdown options in Task.Supervisor.start_link/1 are deprecated. Please pass those options on start_child/3 instead
  (elixir 1.10.1) lib/task/supervisor.ex:86: Task.Supervisor.start_link/1
  (stdlib 3.11) supervisor.erl:379: :supervisor.do_start_child_i/3
  (stdlib 3.11) supervisor.erl:365: :supervisor.do_start_child/2
  (stdlib 3.11) supervisor.erl:349: anonymous fn/3 in :supervisor.start_children/2
  (stdlib 3.11) supervisor.erl:1157: :supervisor.children_map/4
```

**Comment by bryanhuntesl** [comment]
https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/task/supervisor.ex#L80-L96

**Comment by bryanhuntesl** [comment]
```
grep -r Task.Supervisor deps lib     
deps/bugsnag/lib/bugsnag.ex:   supervisor(Task.Supervisor, [[name: Bugsnag.TaskSupervisor, restart: :transient]])
```

**Comment by smaximov** [comment]
See #92.

**Comment by nirev** [comment]
fixed in `2.1.0`
